src/main/src/main/java/org/geoserver/security/impl/AbstractRoleStore.java

```diff
@@ -210,6 +210,9 @@ public void disAssociateRoleFromGroup(GeoServerRole role, String groupname) thro
             roles.remove(role);
             setModified(true);
         }
+        if (helper.group_roleMap.get(groupname) != null && roles != null && roles.size() == 0) {
```

Comment:
Surprised the PMD rule on the usage of isEmpty did not catch this one, but to be consistent with the resto of the code:

```suggestion
        if (helper.group_roleMap.get(groupname) != null && roles != null && roles.isEmpty()) {
```